src/frontend/src/optimizer/plan_node/logical_join.rs

```diff
@@ -649,17 +649,6 @@ impl ExprRewritable for LogicalJoin {
     }
 }
 
-fn is_pure_fn_except_for_input_ref(expr: &ExprImpl) -> bool {
-    match expr {
-        ExprImpl::Literal(_) => true,
-        ExprImpl::FunctionCall(inner) => {
-            inner.is_pure() && inner.inputs().iter().all(is_pure_fn_except_for_input_ref)
-        }
-        ExprImpl::InputRef(_) => true,
```

Comment:
This will be changed. Should we simply let InputRef be pure?

Reply:
I think `InputRef` is pure. :thinking: Because it's output result totally depends on its input.

Reply:
Then you should fix the implementation of ImpureAnalyzer.

Reply:
Maybe add some UT